src/hotspot/share/opto/superword.cpp

```diff
@@ -2258,27 +2308,39 @@ void SuperWord::schedule() {
     }
 #endif
     _packset.clear();
-    return;
+    return false;
   }
 
-#ifndef PRODUCT
-  if (is_trace_superword_info()) {
-    tty->print_cr("SuperWord::schedule: memops_schedule:");
-    memops_schedule.dump();
-  }
-#endif
+  // (4) Apply the vectorization, including re-ordering the memops.
+  return apply(memops_schedule);
+}
 
+bool SuperWord::apply(Node_List& memops_schedule) {
   CountedLoopNode* cl = lpt()->_head->as_CountedLoop();
-  phase()->C->print_method(PHASE_SUPERWORD1_BEFORE_SCHEDULE, 4, cl);
+  phase()->C->print_method(PHASE_AUTO_VECTORIZATION1_BEFORE_APPLY, 4, cl);
 
-  // (4) Use the memops_schedule to re-order the memops in all slices.
-  schedule_reorder_memops(memops_schedule);
-}
+  apply_memops_reordering_with_schedule(memops_schedule);
+  phase()->C->print_method(PHASE_AUTO_VECTORIZATION2_AFTER_REORDER, 4, cl);
 
+  adjust_pre_loop_limit_to_align_main_loop_vectors();
+  phase()->C->print_method(PHASE_AUTO_VECTORIZATION3_AFTER_ADJUST_LIMIT, 4, cl);
+
+  bool is_success = apply_vectorization();
+  phase()->C->print_method(PHASE_AUTO_VECTORIZATION4_AFTER_APPLY, 4, cl);
+
+  return is_success;
+}
```

Comment:
Note: refactored out all the "apply" code into a nice list, and do the `print_method` code right there too.